app/plugin.go

```diff
@@ -402,6 +411,52 @@ func (a *App) GetPlugins() (*model.PluginsResponse, *model.AppError) {
 	return resp, nil
 }
 
+// GetPluginPublicKeys returns all public keys listed in the config.
+func (a *App) GetPluginPublicKeys() ([]string, *model.AppError) {
+	return a.Config().PluginSettings.SignaturePublicKeyFiles, nil
+}
+
+// GetPublicKey will return the actual public key saved in the `filename` file.
+func (a *App) GetPublicKey(filename string) ([]byte, *model.AppError) {
+	filename += pluginSignaturePublicKeyFileExtention
+
+	data, err := a.Srv.configStore.GetFile(filename)
+	if err != nil {
+		return nil, model.NewAppError("GetPublicKey", "app.plugin.get_public_key.get_file.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	return data, nil
+}
+
+// AddPublicKey method will add plugin public key to the config.
+func (a *App) AddPublicKey(name string, key io.Reader) *model.AppError {
+	if err := a.writeFile(name+pluginSignaturePublicKeyFileExtention, key); err != nil {
```

Comment:
I have the same question here as why the file extension was removed as above

Reply:
This way file extension will not be exposed to the end-user. We will add `pluginSignaturePublicKeyFileExtention` to avoid collisions and user will know nothing about that, they will pass whatever file they have.